tests/test directory/test_setextwithformatting.md

```diff
@@ -0,0 +1,41 @@
+Title one
+=========
+
+<!--ts-->
+
+
+<!-- Created by https://github.com/ekalinin/github-markdown-toc -->
+<!-- Added by: corey.sciuto, at: Fri Sep 22 20:15:18 EDT 2023 -->
+
+<!--te-->
+
+Blabla...
+
+## This is test for setext-style without formatting
+
+Blabla...
+
+*Title `two`*
+=============
+
+Blabla...
+
+## This is test for setext-style with formatting
+
+Blabla...
+
+# Title `three`
```

Comment:
So I noticed the backtick test is failing. I did have an issue in my regex (too greedy!) but after fixing that, `# this style header` with backticks in it still fails because it just disappears. I am not exactly great at shell scripting, but I was able to figure out how to output the steps of the `sed` command one-by-one to see where it was going wrong. This all runs _before_ the awk script. 

Here's the output after `    sed -e ':a' -e 'N' -e '$!ba' -e 's/\n<\/h/<\/h/g' |`

```
<h1 id="user-content-title-one"><a class="heading-link" href="#title-one">Title one<span aria-hidden="true" class="octicon o
cticon-link"></span></a></h1>




<p>Blabla...</p>
<h2 id="user-content-this-is-test-for-setext-style-without-formatting"><a class="heading-link" href="#this-is-test-for-setex
t-style-without-formatting">This is test for setext-style without formatting<span aria-hidden="true" class="octicon octicon-
link"></span></a></h2>
<p>Blabla...</p>
<h1 id="user-content-title-two"><a class="heading-link" href="#title-two"><em>Title <code>two</code></em><span aria-hidden="
true" class="octicon octicon-link"></span></a></h1>
<p>Blabla...</p>
<h2 id="user-content-this-is-test-for-setext-style-with-formatting"><a class="heading-link" href="#this-is-test-for-setext-s
tyle-with-formatting">This is test for setext-style with formatting<span aria-hidden="true" class="octicon octicon-link"></s
pan></a></h2>
<p>Blabla...</p>
<h1 id="user-content-title-three"><a class="heading-link" href="#title-three">Title <code>three</code>
<span aria-hidden="true" class="octicon octicon-link"></span></a></h1>
<p>Blabla...</p>
<h2 id="user-content-this-is-a-regression-test-for-atx-style"><a class="heading-link" href="#this-is-a-regression-test-for-a
tx-style">This is a regression test for atx-style<span aria-hidden="true" class="octicon octicon-link"></span></a></h2>
<p>Blabla...</p>
<h1 id="user-content-title-four-is-a-particularly-long-title-because-of-wrapping"><a class="heading-link" href="#title-four-
is-a-particularly-long-title-because-of-wrapping">Title four is a particularly long title because of wrapping<span aria-hidd
en="true" class="octicon octicon-link"></span></a></h1>
<p>Blabla...</p>
<h2 id="user-content-this-is-a-test-for-long-titles"><a class="heading-link" href="#this-is-a-test-for-long-titles">This is 
a test for long titles<span aria-hidden="true" class="octicon octicon-link"></span></a></h2>
<p>Blabla...</p>
                                                                                                                            
                                                                                                                            
                                                                                                                            
```

You'll notice the next step will fail `$grepcmd '<h.*id="user-content-[^"]*".*</h[1-6]' |` because it looks like the GitHub API puts a newline before `span` tags if there's a `code` tag in there as well but not an `em` tag. It's not the length of the line generated...just something about this combination.

So I guess the next step is to write another line like `sed -e ':a' -e 'N' -e '$!ba' -e 's/\n<\/h/<\/h/g' |` that moves `span` as well...